---
title: "CV"
author: "Antonio Ahn"
date: ""
output: html_document
---

Residence: Melbourne, Australia  <br>
E-mail: Antonio.Ahn@petermac.org  <br>
Citizenship: New Zealand

# Employment

**Post-doc**. Goel lab. Peter MacCallum Cancer Centre. Melbourne. Australia. 2019 - present.

**Medical Laboratory Scientist**. Department of immunology and infectious diseases. Dunedin hospital. Part-time work (20 hours per week).	2012- 2016

# Education

**Doctor of Philosophy (PhD)**, 2015- 2019
- University of Otago

**Masters in Medical Laboratory Science**, 2014
- University of Otago
- Graduated with Distinction

**Postgraduate Diploma in Medical Laboratory Science**, 2011
- University of Otago
- Graduated with Distinction, University of Otago

**Bachelor of Medical Laboratory Science**, 2007-2010
- University of Otago

# Skills
## Bioinformatics
RNA-sequencing, lncRNA analysis (using genecode annotations and kallisto), Transposable Element (using salmonTE), microarray gene expression, Immune decomposition tools, Methylation 450k, TCGA data analysis (RNA-seq, methylation 450k).

## Wet-lab
Flow Cytometry, Cell culture, siRNA knockdown, Sequenome (DNA methylation), PCR (RT-qPCR, sanger sequencing, nested PCR), phenotypic assays: migration (scratch assay, Bowden chamber), proliferation (MTT assay), live cell imaging (Incucyte).

# Most recent posters

**Constitutive PD-L1 expression is associated with a distinct geneexpression signature related to an aggressive phenotype in melanoma** <br>
QMB cancer satellite conference, August 2018. Won the poster award.
[Poster](https://www.slideshare.net/secret/XxhxZ7hX2lQBb).

**DNA methylation regulates PD-L1 expression by activating endogenous retrovirus elements and the viral mimicry pathway** <br>
Epigenetics User Group Symposium, July 2018. Won the poster award. <br> Dunedin School of Medicine Symposium, June 2018. Won the poster award. [Poster](https://www.slideshare.net/Antonioahn/dna-methylation-regulates-pdl1-expression-by-activating-endogenous-retrovirus-elements-and-the-viral-mimicry-pathway).

**Stratification of  TCGA melanoma patients according to Tumor Infiltrative CD8 Lymphocytes and PD-L1 expression using RNA-seq and 450k methylation data** <br>
Queenstown Molecular Biology conference, September 2017. Won the speaker award. [Poster](https://www.slideshare.net/Antonioahn/stratification-of-tcga-melanoma-patients-according-to-tumor-infiltrative-cd8-lymphocytes-and-pdl1-expression-using-rnaseq-and-450k-methylation-data).

# Research Experience

**Goel lab** <br>
Peter MacCallum Cancer Centre, 2019 - Present <br>

**Eccles lab and Chatterjee lab** <br>
Pathology Department, University of Otago, 2016 - 2019 <br>

* Involved in finding that a marked global DNA hypomethylation is associated with constitutive PD-L1 expression in melanoma. As part of my PhD project, I characterised melanoma cell lines for constitutive PD-L1 expression using flow cytometry and by analysing RNA-seq and microarray data. Moreover TCGA data was explored to support that constitutive PD-L1 expression is associated with a viral mimicry response. [Publication](https://www.cell.com/iscience/fulltext/S2589-0042(18)30076-2). On-going work includes exploring transposable elements and lncRNA. 

* Described a step-by-step workflow on how to perform RNA-sequencing, starting from from wet-lab considerations to bioinformatics data analysis. Here i was involved in describing the aspect of data processing and differential expression analysis. [Publication](https://link.springer.com/protocol/10.1007/978-1-4939-7834-2_3).

**Eccles lab** <br>
Pathology Department, University of Otago, 2014 - Present <br>

* Reviewed the literature into the characterisation of a slow cycling phenotype which plays a key role in treatment resistance in melanoma. It has been thought that inhibiting tumor cell growth is a favorable outcome, and analogous to tumor cell killing, however increasing studies have indicated that tumor cells with a slow-cycling phenotype can be metabolically active and highly aggressive, with increased potential to form new tumors and metastasize.  [Publication](http://mct.aacrjournals.org/content/16/6/1002).

* Studied the role of DNA methylation in melanoma metastasis. Genome wide methylation sequencing was performed in three cutaneous primary and metastatic melanoma cell line pairs. We identified numerous epigenetic changes from primary to metstatsis and thereby a potential role in metastasis. We then focused on the *EBF3* transcripton factor to determine its role in metastasis. Here i was involved in performing RT-qPCR and various phenotypic assays which included scratch assays, live-cell imaging (incucyte), boyden chamber cell migration and invasion assay and MTT assay. [Publication](http://www.oncotarget.com/index.php?journal=oncotarget&page=article&op=view&path[]=14042&path[]=44772). This work led to evaluating Sequenom EpiTyper platform for validating whole genome methylation sequencing data. [Publication](https://www.futuremedicine.com/doi/10.2217/epi-2016-0176).

* As part of my masters project, we determined the prevalence of recurrent mutations (that have been identified as "drivers" in cancer progression) in the New Zealand melanoma population. Mutations were assessed using Sequenom MelaCarta MassARRAY and Sanger sequencing. Mutations of interest consisted those that reside in *BRAF*, *NRAS*, *RAC1*, *TERT* and *EPHB6*. Here i performed both standard and laser microdissection techniques to isolate the tumor tissue from the surrounding normal tissue. DNA was extracted and sanger sequencing was performed. Moreover, i was involved analysing the Sequnome and sanger sequencing data to determine the prevalence of each mutations. [Publication](http://www.oncotarget.com/index.php?journal=oncotarget&page=article&op=view&path[]=9351&path[]=28900).

* Investigated the "Genetic switch" model in melanoma. This model extends on the "phenotype switch" model which describes that melanoma progression is driven by a contiuous back and forth switch between a proliferative and invasvie phenotype. This switch is regulated by (epi)genetic intrinisc cues and signals from the tumor microenvironment. The "Genetic switch" model describes a *MITF* and *PAX3* as prominent players in this phenotype switch. Here i was involved in performing RT-qPCR. [Publication](https://www.frontiersin.org/articles/10.3389/fonc.2013.00229/full).

* Reviewed the literature into the role of targeted therapy in the adjuvant setting for melanoma. [Publication](https://www.frontiersin.org/articles/10.3389/fonc.2013.00205/full).


**Slatter lab** <br>
Pathology Department, University of Otago, 2011 - 2013 <br>

* I was involved in determining that a C500G SNP (rs11515) is associated with more aggressive breast tumors. This SNP was also associated with an increased expression of a long non-coding RNA called *ANIRL* and a lower expression of p16INK4a (CDKN2A). Furthermore we found that the same SNP is more frequent in a subtype of GMB with no defined telomerase maintainance mechanisms and is associated with poorer survival. In both projects, I contriubuted in genotyping the C500G SNP within *CDKN2A* 3'UTR region from DNA extracted from breast tumors. [Publication1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4720739/). [Publication2](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0026737). 


